providers/account.rb

```diff
@@ -169,7 +169,24 @@ def home_ssh_dir_resource(exec_action)
 
 def authorized_keys_resource(exec_action)
   # avoid variable scoping issues in resource block
-  ssh_keys = Array(new_resource.ssh_keys)
+  ssh_keys = []
+  Array(new_resource.ssh_keys).each do |item|
+    # if item is a valid ssh key, copy it.
+    if item =~ /^(ssh-(dss|rsa|ed25519)|ecdsa-sha2-\w+) AAAA/
```

Comment:
I was referring to shell-style comments, so that updated regular expression should take care of it.

And WRT the example RegExp I gave, I was trying to match any comment line with (or without) leading whitespace. Overkill.

I just realized that doing this RegExp match would also break the ability to use this resource with the `PermitUserEnvironment` functionality of OpenSSH:

* http://www.screenage.de/blog/2012/02/10/how-to-log-history-and-logins-from-multiple-ssh-keys-under-one-user-account-with-puppet/